.github/workflows/start-release.yml

```diff
@@ -69,6 +69,12 @@ jobs:
         env:
           SEMGREP_RELEASE_NEXT_VERSION: ${{ steps.next-version.outputs.next-version }}
         run: make release
+      - name: Update Changelog
```

Comment:
I think you'll need to add a setup-python step above.

Reply:
This worked as-is in a test repo - is there any benefit to adding a setup-python step? I am happy to do so, but didn't wanna cause extra bloat in the workflow.

Reply:
- pins python version to avoid breakage when the GHA runners change it
- adds caching for pipenv deps for faster workflows

Reply:
will test this out, then update the PR. Thanks!